legacy/reqcounter/reqcounter_test.go

```diff
@@ -0,0 +1,127 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reqcounter_test
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	rc "sigs.k8s.io/k8s-container-image-promoter/legacy/reqcounter"
+)
+
+// defaultTime should be used as a timestamp for all RequestCounter objects.
+// The actual time represents: September 22, 2002 at 05:03:16.
+var defaultTime, _ = time.Parse("020106 150405", "020106 150405")
+
+// NewRequestCounter returns a RequestCounter with the given number of requests.
+// All other object fields are set to default values.
+func NewRequestCounter(requests uint64) rc.RequestCounter {
+	return rc.RequestCounter{
+		Mutex:    sync.Mutex{},
+		Requests: requests,
+		Since:    defaultTime,
+		Interval: time.Second,
+	}
+}
+
+func TestInit(t *testing.T) {
+	rc.Init()
+	// Ensure request counting is enabled.
+	require.True(t, rc.EnableCounting, "Init did not enable counting.")
+	// Ensure counters are created.
+	require.Greater(t, len(rc.Monitor.Counters), 0, "Init did not create any request counters within the global Monitor.")
+	// Ensure at least one counter uses the MeasurementWindow.
+	found := false
+	for _, counter := range rc.Monitor.Counters {
+		if counter.Interval == rc.MeasurementWindow {
+			found = true
+			break
+		}
+	}
+	require.True(t, found, "No request counters are using the Interval: MeasurementWindow.")
+}
+
+func TestIncrement(t *testing.T) {
+	// Create counters which use these request counts and
+	// populate the Monitor global variable.
+	counters := []rc.RequestCounter{
+		NewRequestCounter(0),
+		NewRequestCounter(9),
+		NewRequestCounter(2839),
+	}
+	netMonitor := &rc.NetworkMonitor{
+		Counters: []*rc.RequestCounter{
+			&counters[0],
+			&counters[1],
+			&counters[2],
+		},
+	}
+	// Create the monitor counter we expect to get after calling rc.Increment.
+	expectedCounters := []rc.RequestCounter{
+		NewRequestCounter(1),
+		NewRequestCounter(10),
+		NewRequestCounter(2840),
+	}
+	expectedNetMonitor := &rc.NetworkMonitor{
+		Counters: []*rc.RequestCounter{
+			&expectedCounters[0],
+			&expectedCounters[1],
+			&expectedCounters[2],
+		},
+	}
+	// Set the global network monitor.
+	rc.Monitor = netMonitor
+	// Ensure counter modification can only occur when counting is enabled. Therefore,
+	// the global network monitor should not be mutated with this call to Increment.
+	rc.EnableCounting = false
+	rc.Increment()
+	require.EqualValues(t, netMonitor, rc.Monitor, "Counters were modified while counting was disabled.")
+	// Ensure the Increment function actually increments each counter's requests field.
+	rc.EnableCounting = true
+	rc.Increment()
+	require.EqualValues(t, expectedNetMonitor, rc.Monitor, "All counters were not incremented correctly.")
```

Comment:
Reword `All counters` -> `Request counters` because if the Increment() function misbehaves, it may be that some counters were incremented, just not all of them.